iceoryx_hoofs/include/iceoryx_hoofs/memory/relative_pointer.hpp

```diff
@@ -207,6 +198,69 @@ using UntypedRelativePointer = RelativePointer<void>;
 /// @return the static pointer repository
 PointerRepository<segment_id_underlying_t, UntypedRelativePointer::ptr_t>& getRepository() noexcept;
 
+/// @brief Checks if lhs and rhs point to the same pointee
+/// @param[in] lhs Relative Pointer
+/// @param[in] rhs Is the pointer whose pointee is compared with this' pointee
+/// @return True if the pointees are equal, otherwise false
+// AXIVION Next Construct AutosarC++19_03-A13.5.5 : The RelativePointer shall explicitly be comparable to raw pointers
+template <typename T>
+bool operator==(const RelativePointer<T> lhs, T* const rhs) noexcept;
+
+/// @brief Checks if lhs and rhs point to the same pointee
+/// @param[in] lhs Is the pointer whose pointee is compared with this' pointee
+/// @param[in] rhs Relative Pointer
+/// @return True if the pointees are equal, otherwise false
+// AXIVION Next Construct AutosarC++19_03-A13.5.5 : The RelativePointer shall explicitly be comparable to raw pointers
+template <typename T>
+bool operator==(T* const lhs, const RelativePointer<T> rhs) noexcept;
+
+/// @brief Checks if rhs is a nullptr
+/// @param[in] lhs nullptr
+/// @param[in] rhs Relative Pointer
+/// @return True if the pointees are equal, otherwise false
+// AXIVION Next Construct AutosarC++19_03-A13.5.5 : The RelativePointer shall explicitly be comparable to nullptrs
+template <typename T>
+bool operator==(std::nullptr_t, const RelativePointer<T> rhs) noexcept;
+
+/// @brief Checks if lhs is a nullptr
+/// @param[in] lhs Relative Pointer
+/// @param[in] rhs nullptr
+/// @return True if the pointees are equal, otherwise false
+// AXIVION Next Construct AutosarC++19_03-A13.5.5 : The RelativePointer shall explicitly be comparable to nullptrs
+template <typename T>
+bool operator==(const RelativePointer<T> lhs, std::nullptr_t rhs) noexcept;
+
+/// @brief Checks if lhs and rhs point not to the same pointee
+/// @param[in] lhs Relative Pointer
+/// @param[in] rhs Is the pointer whose pointee is compared with this' pointee
+/// @return True if the pointees are not equal, otherwise false
+// AXIVION Next Construct AutosarC++19_03-A13.5.5 : The RelativePointer shall explicitly be comparable to raw pointers
+template <typename T>
+bool operator!=(const RelativePointer<T> lhs, T* const rhs) noexcept;
+
+/// @brief Checks if lhs and rhs point not to the same pointee
+/// @param[in] lhs Is the pointer whose pointee is compared with this' pointee
+/// @param[in] rhs Relative Pointer
+/// @return True if the pointees are not equal, otherwise false
+// AXIVION Next Construct AutosarC++19_03-A13.5.5 : The RelativePointer shall explicitly be comparable to raw pointers
+template <typename T>
+bool operator!=(T* const lhs, const RelativePointer<T> rhs) noexcept;
+
+/// @brief Checks if rhs is a nullptr
+/// @param[in] lhs nullptr
+/// @param[in] rhs Relative Pointer
+/// @return True if the pointees are equal, otherwise false
```

Comment:
```suggestion
/// @return True if rhs is equivalent to a null pointer, otherwise false
```
Equivalent to a null pointer means that it points to nothing.

Similar for other null pointer comparisons.